src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.cpp

```diff
@@ -131,6 +114,18 @@ void ShenandoahDirectCardMarkRememberedSet::mark_range_as_clean(HeapWord* p, siz
   }
 }
 
+void ShenandoahDirectCardMarkRememberedSet::mark_read_table_as_clean() {
+  CardValue* read_table = _card_table->read_byte_map();
+  CardValue* bp = &(read_table)[0];
+  CardValue* end_bp = &(read_table)[_card_table->last_valid_index()];
+
```

Comment:
Looks like there may be an off-by-one error here, or a bad name for "last_valid_index()".  The following loop should continue while bp <= end_bp.

Reply:
Good catch, thanks!